Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue9525.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+
+#if UITEST
+using Xamarin.Forms.Core.UITests;
+using Xamarin.UITest;
+using NUnit.Framework;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+#if UITEST
+	[Category(UITestCategories.ManualReview)]
+#endif
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 9525, "MediaElement Disposing exception when MainPage is changed on iOS", PlatformAffected.iOS)]
+	public class Issue9525 : TestNavigationPage
+	{
+		protected override void Init()
+		{
+#if APP
+			Device.SetFlags(new List<string>(Device.Flags ?? new List<string>()) { "MediaElement_Experimental" });
+
+			PushAsync(CreateRoot());
+#endif
+		}
+		private ContentPage CreateRoot()
+		{ 
+			var button = new Button
+			{
+				AutomationId = "Issue9525Button",
+				Text = "Go to new page",
+			};
+			button.Clicked += Button_Clicked;
+			return new ContentPage
+			{
+				Content = new StackLayout
+				{
+					Children =
+					{
+						new MediaElement
+						{
+							AutomationId = "Issue9525MediaElement",
+							Source="https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4",
```

Comment:
We need to find a video that won't disappear on us. Something uploaded to the source, perhaps.